composer/loggers/mosaicml_logger.py

```diff
@@ -96,10 +115,63 @@ def log_hyperparameters(self, hyperparameters: Dict[str, Any]):
     def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> None:
         self._log_metadata(metrics)
 
+    def log_analytics(self, state: State) -> None:
+        if self.analytics_data is None:
+            return
+
+        metrics: Dict[str, Any] = {
+            'composer/autoresume': self.analytics_data.autoresume,
+            'composer/precision': state.precision,
+        }
+        metrics['composer/eval_loaders'] = []
+        for evaluator in state.evaluators:
+            dataloader = evaluator.dataloader.dataloader
+            if isinstance(dataloader, torch.utils.data.DataLoader):
+                metrics['composer/eval_loaders'].append(evaluator.label)
+
+        metrics['composer/optimizers'] = [{
+            optimizer.__class__.__name__: optimizer.defaults,
+        } for optimizer in state.optimizers]
+        metrics['composer/algorithms'] = [algorithm.__class__.__name__ for algorithm in state.algorithms]
+
+        metrics['composer/loggers'] = [logger.__class__.__name__ for logger in self.analytics_data.loggers]
+
+        # Take the service provider out of the URI and log it to metadata. If no service provider
+        # is found (i.e. backend = ''), then we assume 'local' for the cloud provider.
+        if self.analytics_data.load_path is not None:
+            backend, _, _ = parse_uri(self.analytics_data.load_path)
+            metrics['composer/cloud_provided_load_path'] = backend if backend else 'local'
+        if self.analytics_data.save_folder is not None:
+            backend, _, _ = parse_uri(self.analytics_data.save_folder)
+            metrics['composer/cloud_provided_save_folder'] = backend if backend else 'local'
+
+        # Save interval can be passed in w/ multiple types. If the type is a function, then
```

Comment:
Is there some idea for utility of analytics on save interval? Nothing is coming to mind for me

Reply:
this was requested in a [Slack thread](https://databricks.slack.com/archives/C02EB3VCYRK/p1709073055904539?thread_ts=1709067237.747639&cid=C02EB3VCYRK) a while back. this included some less-helpful metrics (i.e. `num_workers`) so if it isn't useful i'll take it out.

Reply:
yeah i think just drop this one